lib/i18n/tests/interpolation.rb

```diff
@@ -118,6 +118,21 @@ module Interpolation
         assert_raises(I18n::ReservedInterpolationKey) { interpolate(:interpolate) }
       end
 
+      test "interpolation: it does not raise I18n::ReservedInterpolationKey for escaped variables" do
+        assert_nothing_raised do
+          assert_equal '%{separator}', interpolate(:foo => :bar, :default => '%%{separator}')
+        end
+
+        assert_nothing_raised do
+          assert_equal '%%{scope}', interpolate(:default => '%%{scope}')
```

Comment:
The result of this interpolation and the one below it confused me, shouldn't it be `"%{scope}"` like above?

Reply:
To be clear, the interpolation result was not changed by this PR. It keeps the extra `%` even with i18n version 1.14.4 and i18n version 1.14.1.

This means that this PR alone so far is not enough to be able to fix i18n-tasks - when the interpolation is escaped the escape character (the extra `%`) shows up in the output. And when it's not escaped then the ReservedInterpolationKey error is raised.

Reply:
Thanks for doing this - I had a similar PR in progress.

I think that last observation is correct ... we need a change on the `i18n` side to not raise on escaped keyword usage, and a change on the `i18n-tasks` side to escape the keywords when they want literal output as part of the help commands.

Reply:
Yes, I have the `i18n-tasks` change locally.

But due to the escape character not getting removed, it results in the `%` being duplicated, which is not desired: 
```
Usage: i18n-tasks add-missing [options] [locale ...]
    -v, --value VALUE                Value. Interpolates: %%{value}, %%{human_key}, %%{key}, %%{default}, %%{value_or_human_key}, %%{value_or_default_or_human_key}.
 ```

The escape character being kept is inconsistent with how [sprintf](https://ruby-doc.org/3.2.2/format_specifications_rdoc.html) functions, which I assume this formatting is inspired by:
```
irb(main):004:0> sprintf("Value. Interpolates: %%{value}, %%{human_key}, %%{key}, %%{default}, %%{value_or_human_key}, %%{value_or_default_or_human_key}.")  
   
=> "Value. Interpolates: %{value}, %{human_key}, %{key}, %{default}, %{value_or_human_key}, %{value_or_default_or_human_key}."
```

I see interpolation getting skipped (and thus the escape character not getting removed) if no substitution are provided [here](https://github.com/ruby-i18n/i18n/blob/74258ad03021c3592302438b86e08ab778e85007/lib/i18n/backend/base.rb#L194), [here](https://github.com/ruby-i18n/i18n/blob/74258ad03021c3592302438b86e08ab778e85007/lib/i18n/backend/base.rb#L58-L67) and potentially also [in the compiler](https://github.com/ruby-i18n/i18n/blob/74258ad03021c3592302438b86e08ab778e85007/lib/i18n/backend/interpolation_compiler.rb#L100-L103).

Reply:
I wrote tests and this looks to be going deeper than I expected so I made a separate issue: #689

Reply:
This was determined to be intended behaviour in the above linked issue. I added an explanatory comment to the test.